### PR TITLE
Various cleanups for package.json

### PR DESCRIPTION
**Pull request by lindner**
- Add license file and header. Use the BSD 3 clause, same as Polymer.
- Add metadata about issues and source repos.
- Add an engines config to match up with the requirements in README.md
  requiring 9.4.0 or higher.

Addresses #1724

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2014 The Polymer Authors. All rights reserved.
```
Comment (shaper) [inline LICENSE]: Probably want 2017 here? Maybe it's 2016, @seefeldb would know.

File: devtools/package.json [hunk 2]
```diff
@@ -34,5 +34,15 @@
   },
   "scripts": {
     "postinstall": "cross-env empathy install -a deps --ignore diff vis web-animations-js && cross-env tools/css-module-wrap ./node_modules/vis/dist/vis-timeline-graph2d.min.css"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/PolymerLabs/arcs.git"
+  },
+  "author": "The Polymer Project Authors",
+  "license": "BSD-3-Clause",
```
Comment (shaper) [inline devtools/package.json]: We should double check with @seefeldb on the license here and below.

File: package.json [hunk 3]
```diff
@@ -48,5 +48,18 @@
...
+    "url": "https://github.com/PolymerLabs/arcs.git"
+  },
+  "author": "The Polymer Project Authors",
+  "license": "BSD-3-Clause",
+  "homepage": "https://github.com/PolymerLabs/arcs",
+  "bugs": {
+    "url": "https://github.com/PolymerLabs/arcs/issues"
+  },
+  "engines": {
+    "node": ">= 9.4.0"
```
Comment (shaper) [inline package.json]: We had logic somewhere else checking this and alerting user if their version was insufficient, if this starts enforcing it maybe we can remove that logic? I think it's here: https://github.com/PolymerLabs/arcs/blob/master/tools/sigh.js#L99

Comment (lindner) [reply package.json]: Digging into this a little deeper reveals that the engines property is used by the package consumer, it's informational only.  However it does make a good source of truth. So I adjusted the semvar checks to use the canonical versions there.

**Pull request merged**